changelog/unreleased/pull-3776

```diff
@@ -0,0 +1,8 @@
+Bugfix: Fix handling of `maxKeys` in `SearchKey`
```

Comment:
This is way too technical for the average user to understand. What about something like
`Bugfix: Limit number of key files tested while opening a repository`